src/Compilers/CSharp/Portable/FlowAnalysis/UnassignedVariablesWalker.cs

```diff
@@ -48,26 +49,26 @@ private new HashSet<Symbol> Analyze(ref bool badRegion)
             return _result;
         }
 
-        protected override void ReportUnassigned(Symbol symbol, SyntaxNode node)
+        protected override void ReportUnassigned(Symbol symbol, SyntaxNode node, int slot, bool skipIfUseBeforeDeclaration)
         {
             // TODO: how to handle fields of structs?
             if (symbol.Kind != SymbolKind.Field)
             {
                 _result.Add(symbol);
             }
+            else
+            {
+                symbol = GetNonFieldSymbol(slot);
+                Debug.Assert((object)symbol != null);
+                _result.Add(symbol);
+                base.ReportUnassigned(symbol, node, slot, skipIfUseBeforeDeclaration);
```

Comment:
The non-field symbol override did not contain a base call. I can file a bug to look into whether this is the right decision, but I think this code is equivalent to the previous.